### ISSUE TITLE
Update of the License

### ISSUE THREAD

**Issue by gkunz** [linsomniac/python-memcached#187]
Hi,

first of all, thank you very much for your continued efforts to maintain this project! 🫶

I have a slightly boring question regarding the license file as current stored in the repo: The current license file contains a (old) version of the Python Foundation License. It turns out that this license has a reference to the SleepyCat license, which was part of Python back then. The Sleepycat license is a very broad copyleft license, which causes a bit of a head-ache for organizations like mine, who care very much about the licenses of open source software.

I believe, however, that neither python-memchached itself nor Python has a dependency on code under the Sleepcat license. This might just be a artifact of using an old version of the PSF license.

Question 1: would it be possible to update the license file to a current version of the PSF license [1][2]?

Question 2: I even think that the PSF license would actually only be applicable to Python itself, but python-memcached could pick any license, such as BSD, Apache, MIT, etc. However, I am aware that changing the license is complex, and given the large number of contributors to this project not advisable, so I'd just like to bring this to your attention.

Of course, please let me know if my understanding is incorrect.

[1] https://github.com/python/cpython/blob/main/LICENSE
[2] https://opensource.org/license/python-2-0/

**Comment by linsomniac** [comment]
Thanks for bringing this up.  I think it's clear that sleepycat has no bearing to the memcache code, so it is reasonable to not reference that.  Picking up the latest PSF license seems like an easy step to do and I believe achieves the original goal of selecting the PSF license, so is easy to immediately select it.

Now, as to question 2: I agree with you that the PSF license is really oriented towards python itself, and may not be the best choice for python-memcached.  I don't recall if I selected the license, or it was selected before I picked up maintainership.  Either way, I'm not sure much more thought went into it than: "Python uses this license so it should be a good choice for a python library".

If we agree that the PSF license isn't the best choice, the question that goes to what would be a better choice.  One possibility would be the GPL, if it makes it easy with the PSF license being "GPL compatible".  For my own code lately I've been using CC0 as just public domain.  I'd be open to something else.

But, as you say, it's more complicated to change the license, I'm not even sure I can get in touch with all contributors to get their approval.

Short form: I'm going to commit changes to the license file to include just a recent PSF license text.
